source/server/filter_chain_manager_impl.h

```diff
@@ -234,34 +282,34 @@ class FilterChainManagerImpl : public Network::FilterChainManager,
   findFilterChainForSourceIpAndPort(const SourceIPsTrie& source_ips_trie,
                                     const Network::ConnectionSocket& socket) const;
 
+  const FilterChainManagerImpl* getOriginFilterChainManager() {
+    ASSERT(origin_.has_value());
```

Comment:
I thought we prefer prefer these precondition-ish ASSERT since the assertion failure is more friendly in debugging mode.
Sure, I will update in the next PR. Thank you for allowing me amending in follow up!